dangerzone/gui/updater.py

```diff
@@ -182,14 +203,30 @@ def get_latest_info(self) -> UpdateReport:
         to the users.
         """
         try:
-            # TODO: Set timeout.
-            info = requests.get(self.GH_RELEASE_URL).json()
+            res = requests.get(self.GH_RELEASE_URL, timeout=self.REQ_TIMEOUT)
         except Exception as e:
-            # TODO:: Handle errors.
-            raise
+            raise RuntimeError(
+                f"Encountered an exception while querying {self.GH_RELEASE_URL}: {e}"
+            )
+
+        if res.status_code != 200:
+            raise RuntimeError(
+                f"Encountered an HTTP {res.status_code} error while querying"
+                f" {self.GH_RELEASE_URL}"
+            )
 
-        version = info["tag_name"].lstrip("v")
-        changelog = markdown.markdown(info["body"])
+        try:
+            info = res.json()
+        except json.JSONDecodeError as e:
+            raise ValueError(f"Received a non-JSON response from {self.GH_RELEASE_URL}")
+
+        try:
+            version = info["tag_name"].lstrip("v")
+            changelog = markdown.markdown(info["body"])
+        except KeyError as e:
+            raise ValueError(
+                f"Missing required fields in JSON response from {self.GH_RELEASE_URL}"
```

Comment:
Or any exception during the parsing of the markdown

Reply:
Actually no, we catch only `KeyError` here. The reason is that the `str(e)` of a `KeyError` message is just the key name. That's why I wanted to catch it here and add a bit more context.

Markdown rendering errors are actually caught by the catch-all handler and are reported as is.

Reply:
OK. Sounds good.